File: 240404/그 요일은/the-day-of-the-day.java
import java.util.Scanner;
public class Main {
    public static void main(String[] args) {
        // 여기에 코드를 작성해주세요.
        Scanner sc = new Scanner(System.in);
        int m1 = sc.nextInt();
        int d1 = sc.nextInt();
        int m2 = sc.nextInt();
        int d2 = sc.nextInt();
        String date = sc.next();
        int[] count = new int[] {0,0,0,0,0,0,0};
        int[] arr = new int[]{0, 31,29, 31, 30,31,30,31,31,30,31,30,31};
        while(true) {
            if(m1 == m2 && d1 == d2) 
                break;
            count[d1%7]++;
            d1++;
            if(d1 > arr[m1]) {
                d1 = 1;
                m1++;
            }
        }
        if(date.equals("Mon")) {
            System.out.println(count[0]);
        } else if(date.equals("Tue")) {
            System.out.println(count[1]);
        } else if(date.equals("Wed")) {
            System.out.println(count[2]);
        } else if(date.equals("Thu")) {
            System.out.println(count[3]);
        } else if(date.equals("Fri")) {
            System.out.println(count[4]);
        } else if(date.equals("Sat")) {
            System.out.println(count[5]);
        } else {
            System.out.println(count[6]);
        }
         
    }
}

File: 240418/그 요일은/the-day-of-the-day.kt
import java.util.Scanner

fun main() {
    val sc = Scanner(System.`in`)

    val m1 = sc.nextInt()
    val d1 = sc.nextInt()
    val m2 = sc.nextInt()
    val d2 = sc.nextInt()
    val a = sc.next()

    var result1 = (1..m1 - 1).map { monthToDay(it) }.sum() + d1
    val result2 = (1..m2 - 1).map { monthToDay(it) }.sum() + d2

    when (a) {
        "Mon" -> result1 += 0
        "Tue" -> result1 += 1
        "Wed" -> result1 += 2
        "Thu" -> result1 += 3
        "Fri" -> result1 += 4
        "Sat" -> result1 += 5
        else -> result1 += 6
    }

    print((result2 - result1) / 7 + 1)
}

fun monthToDay(n: Int): Int {
    return when(n) {
        1 -> 31
        2 -> 29
        3 -> 31
        4 -> 30
        5 -> 31
        6 -> 30
        7 -> 31
        8 -> 31
        9 -> 30
        10 -> 31
        11 -> 30
        else -> 31
    }
}